lib/src/endpoints/endpoint_paging.dart

```diff
@@ -42,29 +46,40 @@ class Page<T> {
   int get nextOffset => _paging.offset + _paging.limit;
 }
 
-class Pages<T> {
-  static const defaultLimit = 20;
+const defaultLimit = 20;
 
+abstract class _Pages<T> {
   SpotifyApiBase _api;
   String _path;
-  ParserFunction<T> _pageMapper;
-  ParserFunction<Object> _pageContainerMapper;
   String _pageKey;
-  List<Page<T>> _bufferedPages = [];
-  bool _cancelled = false;
-
-  Pages(this._api, this._path, this._pageMapper,
-      [this._pageKey = null, this._pageContainerMapper = null]) {
-    if (_pageKey != null && _pageContainerMapper == null) {
-      throw new ArgumentError.notNull('_pageContainerMapper');
-    } else if (_pageKey == null && _pageContainerMapper != null) {
-      throw new ArgumentError.notNull('_pageKey');
+  ParserFunction<Object> _pageContainerParser;
+
+  _Pages(this._api, this._path,
+      [this._pageKey = null, this._pageContainerParser = null]) {
+    if (_pageKey != null && _pageContainerParser == null) {
+      throw new ArgumentError.notNull('pageContainerParser');
+    } else if (_pageKey == null && _pageContainerParser != null) {
+      throw new ArgumentError.notNull('pageKey');
     }
   }
 
-  Future<Page<T>> first([int limit = defaultLimit]) {
-    return _getPage(limit, 0);
+  /// Abstract method that is used to do the api call and json serializing
+  Future<T> call(int limit, int offset);
```

Comment:
```suggestion
  Future<T> getPage(int limit, int offset);
```

The name `getPage` more clearly describes the intent of this function. Not to mention it offers a solution for end-users to get a specific page. (e.g. Getting the next page after calling `Pages.first`)

In hindsight, not having a public `getPage` was an oversight on my part. 😝